### DISCUSSION TITLE
Error with wb_add_plot() and wb_dims()

### DISCUSSION THREAD

**Discussion by FrancisNoblat** [JanMarvin/openxlsx2#1145] [Q&A]
Hey, Jan, and all contributors to openxlsx2!

Recently, when trying to add a plot to my tables, I habe been receiving the error message bellow:

```
Warning messages:
1: 'start_row/start_col' ist veraltet.
Benutzen Sie stattdessen 'dims'
Siehe help("Deprecated") und help("openxlsx2-deprecated"). 
2: 'start_col/start_row' ist veraltet.
Benutzen Sie stattdessen 'dims'
Siehe help("Deprecated") und help("openxlsx2-deprecated").
```

However, following the instructions given, and using the wb_dims() command (as per the _„Insert the current plot into a worksheet“_ and the _„Helper to specify the dims argument“_ reference pages), the same error message returns.
For reference, I am writting as follows:

```
wb <- wb_add_plot(wb, dims = "A1", width = 5, height = 5)
```

To which not only the error message is displayed, but the plots, when manually checking the excell file, are added to the A3 cell.
I have also tried with the

```
wb_dimensions <- wb_dims(from_row= 1, from_col= 1)
wb <- wb_add_plot(wb, dims = wb_dimensions, width = 5, height = 5)
```

but also without success.

Is there any known issue regarding the wb_add_plot() command, or any solution to circunvent such errors?

Once again, thank you all for your answer!

**Comment by JanMarvin** [comment]
Hi @N-bl-t , thanks, strange indeed. You could try to remove width and height and specify these as „A1:D5“ or something like this. But using width and height with a single cell reference should work as well.
Could you have a look @olivroy ? I’m currently traveling without a computer. Otherwise this will have to wait until next week.

**Reply by olivroy** [reply]
Hi! There doesn't seem to be an issue with `wb_dims()`. You have to print your ggplot object to the R console before adding it to a workbook.


```r

wb <- wb_workbook()
wb$add_worksheet()
library(ggplot2)
 p2 <- ggplot(Orange, aes(x = age, y = circumference, color = Tree)) +
   geom_point() + geom_line()
 print(p2)
 wb_dimensions <- wb_dims(from_row= 1, from_col= 1)
# reassign wb
wb <- wb_add_plot(
  wb,
  dims = wb_dimensions,
  width = 5, height = 5
)
# preview
wb$open()
```

https://janmarvin.github.io/openxlsx2/articles/openxlsx2_charts_manual.html or [`?wb_add_plot`](https://janmarvin.github.io/openxlsx2/reference/wb_add_plot.htm)

**Reply by JanMarvin** [reply]
Thank you!

**Reply by FrancisNoblat** [reply]
Hello, and thank you @JanMarvin, @olivroy for the quick reply!

Unsure how I can make the error reproductible, as to allow you to better evaluate its possible source, it unfortunately remains as an output for my script.

I am also unsure whether it could be the result of an interaction with other packages, the context of the generation of my graphics and tables is through a loop, using the package „RDCOMClient“, together with „ggplot2“ and „ggrepel“.
Least but not least, it’s still curious that even though the insertion doesn’t happen as intended, it remains still being pasted to A3, doesn’t matter how I try to change the wb_dims() parameters.

If not a single cause can be pinpointed in the meantime, waiting until next week for a better assessment would be of absolutely no concern.

That being said, thank you again!

**Reply by JanMarvin** [reply]
Well, maybe start with baby steps, does Olivier’s example work for you?

**Reply by olivroy** [reply]
If you are able to share a minimal reproducible example, we may need to help.. If you say you make a loop, maybe adding a `Sys.sleep(0.5)` may help too. If you attempt to put multiple plots in the same place, it is possible that the replacement is not correctly placed / overriden too.

**Reply by FrancisNoblat** [reply]
It did work, indeed.

Which makes me think whether it is a result of a specific dataset when converted to plot issue, especially when dealing with more complex plotting, for it also happens form e when using the wb_add_image() function.

However, it still baffles me that it has A3 as a „fixed“ paste-cell, which becomes even more curious when clicking, then it activates a copy-and-drag function in excell.

As for the @olivroy multiple plots in the same place point, that is not the case, for each graphic in the workbook is set to be at a different cell:

```
plot <- get(paste0("gra_dl_sk",gra_kl))
plot
wb <- wb_add_plot(wb, dims = "A1", width = 7.5, height = 7.5)
```

but then, again, the error shows up (and in the resulting file it remains still being pasted to A3), doesn’t matter how I try to change the wb_dims() parameters.

Either way, will be anxiously waiting for next week, and an eventual follow-up!

**Reply by JanMarvin** [reply]
My assumption is that something else is interfering. Maybe a typo and some `A3` dim is pulled in? The warning should occur only if non dims references are used. Maybe something forgotten, when updating?

Happy to help, but I’m afraid, you’d have to work on the reproduceable example

**Reply by JanMarvin** [reply]
Hi @N-bl-t , do you still have the issue? I had a quick look, but the only references to "A3" in our code are related to pivot tables and unlikely to be pulled into this.

> However, it still baffles me that it has A3 as a „fixed“ paste-cell, which becomes even more curious when clicking, then it activates a copy-and-drag function in [Excel].

Not sure what this is. For reference, does it work without `RDCOMClient`?
